presto-memsql/src/main/java/io/prestosql/plugin/memsql/MemSqlClient.java

```diff
@@ -88,6 +149,116 @@ protected String getTableSchemaName(ResultSet resultSet)
         return resultSet.getString("TABLE_CAT");
     }
 
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        String jdbcTypeName = typeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new PrestoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        Optional<ColumnMapping> unsignedMapping = getUnsignedMapping(typeHandle);
+        if (unsignedMapping.isPresent()) {
+            return unsignedMapping;
+        }
+
+        if (jdbcTypeName.equalsIgnoreCase("json")) {
+            return Optional.of(jsonColumnMapping());
+        }
+
+        int columnSize = typeHandle.getColumnSize();
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+                VarcharType varcharType = (columnSize <= VarcharType.MAX_LENGTH) ? createVarcharType(columnSize) : createUnboundedVarcharType();
+                // Remote database can be case insensitive.
```

Comment:
can we know if it is and react accordingly?

Reply:
We can fire a query to test the easy cases (eg `A = a`) but it is possible for such checks to pass for ASCII characters but fail for unicode characters (eg. `S = ß`) depending on the collation used by the remote database.

I'm working on adding a smoke test to at-least verify that such-cases are caught in tests.